tests/integration/test_legacy_mysql.py

```diff
@@ -0,0 +1,208 @@
+
+
+@pytest.mark.order(1)
+@pytest.mark.skip_if_deployed
+@pytest.mark.legacy_mysql_tests
+async def test_build_and_deploy(ops_test: OpsTest) -> None:
+    """Build and deploy the mysql charm.
+
+    Args:
+        ops_test: The ops test framework
+    """
+    async with ops_test.fast_forward():
+        # Build and deploy the mysql charm
+        charm = await ops_test.build_charm(".")
+        resources = {"mysql-image": METADATA["resources"]["mysql-image"]["upstream-source"]}
+        config = {
+            "mysql-interface-user": "keystone",
+            "mysql-interface-database": "keystone",
+        }
+        await ops_test.model.deploy(
+            charm,
+            resources=resources,
+            config=config,
+            application_name=DATABASE_APP_NAME,
+            num_units=3,
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[DATABASE_APP_NAME],
+            status="active",
+            raise_on_blocked=True,
+            timeout=1000,
+            wait_for_exact_units=3,
+        )
+        assert len(ops_test.model.applications[DATABASE_APP_NAME].units) == 3
+        for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+            assert unit.workload_status == "active"
+
+
+@pytest.mark.order(2)
+@pytest.mark.abort_on_fail
+@pytest.mark.legacy_mysql_tests
+async def test_osm_keystone_bundle_mysql(ops_test: OpsTest) -> None:
+    """Deploy the osm keystone bundle to test the legacy 'mysql' relation.
+
+    Args:
+        ops_test: The ops test framework
+    """
+    async with ops_test.fast_forward():
+        # Build and deploy the osm-keystone charm
+        # (using ops_test.juju instead of ops_test.deploy as the latter does
+        # not correctly deploy with the correct resources)
+        # TODO: Replace edge channel with stable when osm-keystone is promoted
+        await ops_test.juju(
+            "deploy",
+            "--channel=latest/edge",
+            "--trust",
+            "--resource",
+            "keystone-image=opensourcemano/keystone:testing-daily",
+            "osm-keystone",
+            f"{OSM_KEYSTONE_APP_NAME}",
+        )
+
+        await ops_test.model.wait_for_idle(
+            apps=[OSM_KEYSTONE_APP_NAME],
+            status="blocked",
+            raise_on_blocked=False,
+            timeout=1000,
+            wait_for_exact_units=1,
+        )
+
+        # Relate the mysql charm with the osm-keystone charm
+        await ops_test.model.relate(f"{OSM_KEYSTONE_APP_NAME}:db", f"{DATABASE_APP_NAME}:mysql")
+        await ops_test.model.wait_for_idle(
+            apps=[OSM_KEYSTONE_APP_NAME, DATABASE_APP_NAME],
+            status="active",
+            raise_on_blocked=False,
+            timeout=1000,
+        )
+
+        # Ensure that the keystone database exists
+        show_databases_sql = [
+            "SHOW DATABASES",
+        ]
+
+        random_unit = ops_test.model.applications[DATABASE_APP_NAME].units[0]
```

Comment:
not sure if this unit is actually random
```suggestion
        db_unit = ops_test.model.applications[DATABASE_APP_NAME].units[0]
```